server/api/lib/mavat/index.js

```diff
@@ -39,22 +39,99 @@ const init = () =>
 		})();
 	});
 
-const downloadFile = (url, file, entityDocId, entityDocNumber) => {
+const downloadChallengedFile = (url, file, options) => {
 	return new Promise((resolve) => {
-		http.get(url, (response) => {
-			response.pipe(file);
-			file.on('finish', async function () {
-				await file.close();
-				resolve(true);
-			});
+		options = options || {};
+
+		http.get(url, options, (response) => {
+			if (response.statusCode !== 200) {
+				Log.info(`downloadChallengedFile failed with status ${response.statusCode} for url ${url}`);
+				resolve(false);
+			} else {
+				const contentType = response.headers['content-type'] || '';
+
+				// if content-type is text/html this isn't the file we wish to download but one
+				// of the challenge stages
+				if (contentType.startsWith('text/html')) {
```

Comment:
why not use `contains` rather than `startswith`

Reply:
because the first value in the content-type header is the mime type